### ISSUE TITLE
@link highlighting broken when linking to class

### ISSUE THREAD

**Issue by SamJakob** [redhat-developer/vscode-java#1753]
When using the @link doc tag to refer to a class, rather than a method ([which is valid](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#link)), it is not highlighted, however it is for a method.

##### Environment
- Operating System: Windows 10
- JDK version: JavaSE-1.8
- Visual Studio Code version: 1.52.1
- Java extension version: 0.73.0

##### Steps To Reproduce
1. Open a Java file
2. Paste in the following Javadoc comment
```java
/**
 * {@link String}
 * {@link String#toCharArray()}
 */
```
![image](https://user-images.githubusercontent.com/37072691/103441944-7f7aa600-4c49-11eb-97f2-81979ff4cb12.png)
Notice how only the latter has correct syntax highlighting.

##### Current Result
Syntax highlighting is only applied to the @link tag when referring to a method.

##### Expected Result
Syntax highlighting should be applied to the @link tag when referring to a method or a class.

**Comment by fbricon** [comment]
The textmate grammar is defined in [vscode itself](https://github.com/microsoft/vscode/blob/master/extensions/java/syntaxes/java.tmLanguage.json), but the source of the grammar is actually atom, so the problem should be reported there: https://github.com/atom/language-java/issues

**Comment by SamJakob** [comment]
Filed upstream.

**Comment by 0dinD** [comment]
FYI @fbricon @SamJakob this issue can also be fixed via semantic highlighting. I had this in mind for my next iteration of semantic highlighting improvements, and I've actually already implemented a fix for this specific issue, which has been sitting on my local development branch for a while.

Sadly I got busy and never got to the point of submitting a PR. But I think I'll have time to do that next week. So if all goes well I'm guessing this can make it into the next release of the extension. 

Also, isn't the TextMate grammar defined by the extension itself now, since #1638? I mean, the issue still has to be fixed upstream, but we don't really depend upon VS Code anymore.
